dlrover/python/common/event/reporter.py

```diff
@@ -0,0 +1,286 @@
+# Copyright 2025 The DLRover Authors. All rights reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+from datetime import datetime
+from typing import Dict
+
+from dlrover.python.common.constants import EventReportConstants, NodeStatus
+from dlrover.python.common.log import default_logger as logger
+from dlrover.python.common.node import Node
+from dlrover.python.common.singleton import Singleton
+from dlrover.python.master.elastic_training.net_topology import (
+    NodeTopologyMeta,
+)
+from dlrover.python.scheduler.job import JobArgs
+from dlrover.python.training_event import DLRoverMasterEvent
+from dlrover.python.training_event.emitter import DurationSpan
+
+_master_evt = DLRoverMasterEvent().singleton_instance()
+
+
+class EventReporter(Singleton):
+    @classmethod
+    def inner_report(
```

Comment:
The name `inner_report` is a little confusing.

Reply:
Renamed and annotations added.